application-admintools-default/src/test/java/com/xwiki/admintools/internal/data/identifiers/CurrentServerTest.java

```diff
@@ -0,0 +1,162 @@
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.admintools.internal.data.identifiers;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Named;
+import javax.inject.Provider;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.xwiki.component.phase.InitializationException;
+import org.xwiki.test.annotation.BeforeComponent;
+import org.xwiki.test.junit5.mockito.ComponentTest;
+import org.xwiki.test.junit5.mockito.InjectMockComponents;
+import org.xwiki.test.junit5.mockito.MockComponent;
+
+import com.xwiki.admintools.ServerIdentifier;
+import com.xwiki.admintools.configuration.AdminToolsConfiguration;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit test for {@link CurrentServer}
+ *
+ * @version $Id$
+ */
+@ComponentTest
+public class CurrentServerTest
+{
+    @InjectMockComponents
+    private CurrentServer currentServer;
+
+    @MockComponent
+    private Provider<List<ServerIdentifier>> supportedServers;
+
+    @MockComponent
+    private ServerIdentifier mockServerIdentifier;
+
+    @MockComponent
+    @Named("default")
+    private AdminToolsConfiguration adminToolsConfig;
+
+    @BeforeComponent
+    void setUp(){
+        // Mock the list of supported servers
+        List<ServerIdentifier> mockServerIdentifiers = new ArrayList<>();
+        mockServerIdentifiers.add(mockServerIdentifier);
+        when(supportedServers.get()).thenReturn(mockServerIdentifiers);
+        when(mockServerIdentifier.isUsed()).thenReturn(true);
+    }
+
+    @Test
+    void Initialize() throws InitializationException
+    {
+        // Mock the behavior of adminToolsConfig
+        when(adminToolsConfig.getServerPath()).thenReturn("exampleServerPath");
+
+        // Call the initialize method
+        currentServer.initialize();
+
+        // Verify that the currentServerIdentifier is set correctly
+        assertEquals(mockServerIdentifier, currentServer.getCurrentServer());
+    }
+
+    @Test
+    void updateCurrentServerAfterInitializationNotFound() throws InitializationException
```

Comment:
You should have 2 tests here. One that tests `initializeWithServerNotUsed` and another for `updateCurrentServer`, which will test if the currentServer is updated correctly when `when(serverIdentifier.isUsed()).thenReturn(false);`, and when `when(serverIdentifier.isUsed()).thenReturn(true);` (both cases can be in the same test)

Reply:
I do have the initializeWithServerNotFound test. Do you mean to create two functions and call them in a test?

Reply:
I think I added after on the `initializeWithServerNotFound` test that it is similar to what I described here, so indeed you already have it. 
Now, since you already have the initialize test, why you need to call `initialize` in here? The first 3 lines from this test are just testing again what `initializeWithServerNotFound` does. What you want to test in here is that `updateCurrentServer` will result in a correct `getCurrentServer`. So I was suggesting something like:
```
when(serverIdentifier.isUsed()).thenReturn(true);
currentServer.updateCurrentServer();
assertEquals(serverIdentifier, currentServer.getCurrentServer());

when(serverIdentifier.isUsed()).thenReturn(false);
currentServer.updateCurrentServer();
assertNull(currentServer.getCurrentServer());
```